pyccel/ast/core.py

```diff
@@ -2271,6 +2210,58 @@ def prefix(self):
         """
         return self._prefix
 
+class ConstructorCall(DottedFunctionCall):
+
+    """
+    Represents a Constructor call in the code.
+
+    A node which holds all information necessary to represent a Constructor
+    call in the code.
+
+    Parameters
+    ----------
+    func : FunctionDef, str
+        An instance of FunctionDef or function name.
+
+    arguments : list, tuple, None
+        A list of arguments.
+
+    cls_variable : CustomDataType, optional
+        An instance of `CustomDataType` representing the class variable associated with the constructor.
```

Comment:
I do not understand what you mean by "class variable". An instance of this class?

Looking at the printers it appears that this it the variable on the left-hand side of an assignment, where the right-hand side is a constructor call.

Reply:
Fixed.